go/src/socialapi/workers/payment/subscription.go

```diff
@@ -119,30 +116,45 @@ func EnsureSubscriptionForGroup(groupName string, params *stripe.SubParams) (*st
 		return nil, ErrCustomerNotExists
 	}
 
+	hasSource, err := checkCustomerHasSource(group.Payment.Customer.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if !hasSource {
+		return nil, ErrCustomerSourceNotExists
```

Comment:
I'm not arguing against having the distinction, all I'm saying we could use error values only - the same as we use `mgo.ErrNotFound` for mongo, e.g.:

```go
err := checkCustomerHasSource(...)
if err == ErrCustomerSourceNotExists {
    // handle no-card-case
}
if err != nil {
    // handle not-able-to-check-it-case
}